docs/proposal/20220712-garbage-collection.md

```diff
@@ -0,0 +1,339 @@
+    - [Functional](#functional)
+    - [Non-Functional](#non-functional)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      - [Proposed Changes](#proposed-changes)
+        - [Cluster Creation](#cluster-creation)
+          - [API Changes](#api-changes)
+          - [Controller Changes](#controller-changes)
+          - [New Garbage Collection Service](#new-garbage-collection-service)
+        - [Cluster Deletion](#cluster-deletion)
+          - [Controller Changes](#controller-changes-1)
+          - [New Garbage Collection Service](#new-garbage-collection-service-1)
+        - [clusterawsadm changes](#clusterawsadm-changes)
+    - [Alternative Approaches Considered](#alternative-approaches-considered)
+      - [Using CCM to do the delete](#using-ccm-to-do-the-delete)
+    - [Risks and Mitigations](#risks-and-mitigations)
+      - [Replicating CCM](#replicating-ccm)
+      - [Similar functionality in upstream CAPI](#similar-functionality-in-upstream-capi)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+      - [Alpha](#alpha)
+      - [Beta](#beta)
+      - [Stable](#stable)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+- CAPA - An abbreviation of Cluster API Provider AWS.
+- ELB - Elastic Load Balancer
+- NLB - Network Load Balancer
+- CCM - Cloud Controller Manager
+
+## Summary
+
+If you create a child cluster using CAPA which then in turn creates a `Service` of type `LoadBalancer` this results in a load balancer being created in AWS for that service. The type of load balancer created by default is a **Classic ELB** but you can also create a NLB by annotating your service. For example:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: podinfo-nlb
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
+```
+
+If you try to delete the child/tenant cluster using CAPI/CAPA then it will fail to delete the clusters infrastructure fully in AWS as the VPC is still being using by the NLB that was created. For example:
+
+```text
+E0609 15:49:16.022022  ###### API Changes) b │
+│ efore detaching the gateway.\n\tstatus code: 400, request id: 65dc0fa0-584f-4256-baf5-a2aac2d2dde4" "reconciler group"="controlplane.cluster.x-k8s.io" "reconciler kind"="AWSManaged │
+│ ControlPlane" "name"="capi-managed-test-control-plane" "namespace"="default" 
+```
+
+Currently, CAPA will attempt to delete all the resources it has directly created as part of the cluster lifecycle management. However, if the CCM in the child/tenant cluster has created any resources then these will not be attempted to be deleted.
+
+This proposal outlines a new feature that will be added to CAPA that will delete externally created resources, such as load balancers & security groups, of the child/tenant cluster. This will be referred to as **garbage collection".
```

Comment:
```suggestion
This proposal outlines a new feature that will be added to CAPA that will delete externally created resources, such as load balancers & security groups, of the child/tenant cluster. This will be referred to as **garbage collection**.
```

Reply:
Changed.